prusti-viper/src/encoder/mir_encoder/mod.rs

```diff
@@ -20,13 +20,93 @@ use rustc_middle::{mir, ty};
 use rustc_index::vec::{Idx, IndexVec};
 use rustc_span::{Span, DUMMY_SP};
 use log::{trace, debug};
-use std::collections::HashMap;
+use std::{
+    collections::HashMap,
+    fmt::Display,
+};
 use prusti_interface::environment::mir_utils::MirPlace;
 use downcast_detector::detect_downcasts;
 
 pub static PRECONDITION_LABEL: &'static str = "pre";
 pub static WAND_LHS_LABEL: &'static str = "lhs";
 
+/// Result of encoding a place
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub enum PlaceEncoding {
```

Comment:
Could you move `PlaceEncoding` and its `impl` to a new `mir_encoder/place_encoding.rs` file?